src/creature.cpp

```diff
@@ -460,32 +461,32 @@ void Creature::onDeath()
 {
 	bool lastHitUnjustified = false;
 	bool mostDamageUnjustified = false;
-	Creature* lastHitCreature = g_game.getCreatureByID(lastHitCreatureId);
-	Creature* lastHitCreatureMaster;
+	std::shared_ptr<Creature> lastHitCreature = g_game.getCreatureByID(lastHitCreatureId);
+	std::shared_ptr<Creature> lastHitCreatureMaster;
 	if (lastHitCreature) {
-		lastHitUnjustified = lastHitCreature->onKilledCreature(this);
+		lastHitUnjustified = lastHitCreature->onKilledCreature(getCreature());
 		lastHitCreatureMaster = lastHitCreature->getMaster();
 	} else {
 		lastHitCreatureMaster = nullptr;
 	}
 
-	Creature* mostDamageCreature = nullptr;
+	std::shared_ptr<Creature> mostDamageCreature = nullptr;
```

Comment:
Could this be `const auto`
```suggestion
	const auto mostDamageCreature = std::shared_ptr<Creature>(nullptr);
```